### PR TITLE
feat: failover historic query if deposit is malformed

### PR DESCRIPTION
**Pull request by james-a-morris**
We should allow our queries to _self-heal_ if they detect a malformed deposit. This will check if a deposit returned from the cache is malformed. If it is found to be malformed, we will ignore it and re-cache a correct version.

### CODE REVIEW

File: src/utils/DepositUtils.ts [hunk 1]
```diff
@@ -55,15 +55,21 @@ export async function queryHistoricalDepositForFill(
   let deposit: DepositWithBlock, cachedDeposit: Deposit | undefined;
   if (cache) {
     cachedDeposit = await getDepositInCache(getDepositKey(fill), cache);
+    if (isDefined(cachedDeposit) && !isDepositFormedCorrectly(cachedDeposit)) {
```
Comment (mrice32) [inline src/utils/DepositUtils.ts:61]: Should we also delete it from redis when this happens?

**Review by mrice32 — COMMENTED** [review]
Just one comment!

File: src/utils/DepositUtils.ts (hunk 1 above)
Comment (nicholaspai) [reply src/utils/DepositUtils.ts:61]: I was assuming that setting `cachedDeposit = undefined` was James' way to delete it but if not, then we should explicitly delete it

Comment (james-a-morris) [reply src/utils/DepositUtils.ts:61]: By setting it to undefined here, we eventually re-set the key on line 77

File: src/utils/DepositUtils.ts [hunk 2]
```diff
@@ -55,15 +55,27 @@ export async function queryHistoricalDepositForFill(
...
+      spokePoolClient.logger.warn({
+        at: "[SDK]:DepositUtils#queryHistoricalDepositForFill",
+        message: "Cached deposit was not formed correctly, removing from cache",
+        fill,
+        cachedDeposit,
+      });
+      // By setting this value to undefined, we eventually have to pull
+      // the deposit from our spoke pool client. Because this new deposit
+      // is formed correctly, we will cache it below.
+      cachedDeposit = undefined;
```
Comment (james-a-morris) [inline src/utils/DepositUtils.ts:71]: @mrice32 @pxrl @nicholaspai added a comment to highlight using the existing logic

File: src/utils/DepositUtils.ts (hunk 1 above)
Comment (mrice32) [reply src/utils/DepositUtils.ts:61]: Ah good point.

**Review by mrice32 — APPROVED** [review]

File: src/utils/ValidatorUtils.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const AddressValidator = define<string>("AddressValidator", (v) => ethers.utils.isAddress(String(v)));
+const BigNumberValidator = define<BigNumber>("BigNumberValidator", (v) => ethers.BigNumber.isBigNumber(v));
+
+const DepositSchema = object({
+  depositId: Min(Number(), 0),
+  depositor: AddressValidator,
+  recipient: AddressValidator,
+  originToken: AddressValidator,
+  amount: BigNumberValidator,
+  originChainId: Min(Number(), 0),
```
Comment (nicholaspai) [inline src/utils/ValidatorUtils.ts:14]: Can we make all these `Number` --> `Integer`?

**Review by nicholaspai — APPROVED** [review]

File: src/utils/ValidatorUtils.ts (hunk 3 above)
Comment (james-a-morris) [reply src/utils/ValidatorUtils.ts]: Done: https://github.com/across-protocol/sdk-v2/pull/428/commits/fe9d4251b9ca527bcbf384f1525ed895a8d72730

**Comment by james-a-morris** [comment]
Merging as we have new unit tests that cover real world data

**Pull request merged**